doc/userguide/rules/intro.rst

```diff
@@ -248,10 +251,54 @@ Now, let's say we have a rule with the following header::
 Only the traffic from the client to the server will be matched by this rule,
 as the direction specifies that we do not want to evaluate the response packet.
 
+Now,if we have a rule with the following header::
+
+    alert tcp 1.2.3.4 any <> 5.6.7.8 80
+
+Suricata will duplicate it and use the same rule with headers in both directions :
+
+    alert tcp 1.2.3.4 any -> 5.6.7.8 80
+    alert tcp 5.6.7.8 80 -> 1.2.3.4 any
+
 .. warning::
 
    There is no 'reverse' style direction, i.e. there is no ``<-``.
 
+Bidirectional rules
+~~~~~~~~~~~~~~~~~~~
+
+Here is an example of a bidirectional rule:
+
+.. container:: example-rule
+
+    alert http any any :example-rule-emphasis:`=>` 5.6.7.8 80 (msg:"matching both uri and status"; sid: 1; http.uri; content: "/download"; http.stat_code; content: "200";)
+
+It will match on flows to 5.6.7.8 and port 80.
+And it will match on a full transaction, using both the uri from the request,
+and the stat_code from the response.
+As such, it will match only when Suricata got both request and response.
+
+Bidirectional rules can use direction-ambiguous keywords, by first using
+``bidir.toclient`` or ``bidir.toserver`` keywords.
+
+.. container:: example-rule
+
+    alert http any any => 5.6.7.8 80 (msg:"matching json to server and xml to client"; sid: 1;  :example-rule-emphasis:`bidir.toserver;` http.content_type; content: "json";  :example-rule-emphasis:`bidir.toclient;` http.content_type; content: "xml";)
+
+Bidirectional rules have some limitations :
+
+* They are only meant to work on transactions with first a request to the server,
+  and then a response to the client, and not the other way around (not tested).
```

Comment:
lets add a test for this

Reply:
Lol : I wish I had a test for this :-)

Where can I get a Suricata transaction that is initiated by the server ?
This is untested because I do not know a test case for it...

Reply:
smtp at least has first data from the server generally, not sure if that leads to a tx

Reply:
I guess there are other limitations to consider: async oneside means no matches. Also, midstream means we may pick a tx up on response only? So that wouldn't match either.

This is true for current rules that use flowbits as well of course, but it's good to point it out I think

Reply:
> smtp at least has first data from the server generally

yes

> not sure if that leads to a tx

Indeed it does not :-/

> async oneside means no matches

Yes, should we refuse to load ?

> Also, midstream means we may pick a tx up on response only? So that wouldn't match either.

Indeed, but just like a request with the connection ending before getting an answer would not match

Reply:
Investigated HTTP2 push more closely and I created https://redmine.openinfosecfoundation.org/issues/7317

It looks like a unidirectional tx from server...